docs/mutations.md

```diff
@@ -37,133 +38,233 @@ any outstanding TODOs before opening issues.
+mts.tables.sites.position
+```
+
+Specifying `discrete_genome=False` places mutations at floating point positions
+continously along the genome.
+
+```{code-cell}
+mts = msprime.sim_mutations(ts, rate=0.1, random_seed=5, discrete_genome=False)
+mts.tables.sites.position
 ```
 
+Note that using `discrete_genome=False` is equivalent to specifying an infinite sites
+model.
+
+
 (sec_mutations_time_span)=
 
 ## Restricting time span
 
-```{eval-rst}
-.. todo:: Docs and examples for the start_time and end_time params.
+The time range where mutations can be placed is specified using the `start_time`
+and `end_time` parameters. For instance, the following only allows mutations to occur
+earlier than time ``1`` in the tree:
 
+```{code-cell}
+ts = msprime.sim_ancestry(2, random_seed=1)
+mts = msprime.sim_mutations(ts, rate=2, start_time=1, random_seed=1)
+mts.tables.mutations.time
 ```
 
+Note, however, that the child node of the edge where the mutation occured can be younger
+than `start_time`.
+
+```{code-cell}
+mts.tables.nodes.time[mts.tables.mutations.node]
+```
+
+It is also possible to use multiple calls of {func}`.sim_mutations` with `start_time`
+and `end_time` specified to simulate differing mutation rates at different
+periods of time. For instance, the following code simulates mutations at a low rate
+prior to time ``1`` and at a higher rate more recently.
+
+```{code-cell}
+ts = msprime.sim_ancestry(2, random_seed=1)
+mts = msprime.sim_mutations(ts, rate=0.1, start_time=1, random_seed=1)
+print(mts.tables.mutations.time)
+mts = msprime.sim_mutations(mts, rate=4, end_time=1, random_seed=1)
+print(mts.tables.mutations.time)
+```
+
+As explained in {ref}`the following section <sec_mutations_existing>`, reversing the order of these two
+lines will result in an error, as older mutations must be added first.
+
 (sec_mutations_existing)=
 
 ## Existing mutations
 
-```{eval-rst}
-.. todo:: Docs and examples for what to do when you're adding mutations
-    to a ts that already has mutations. Like what the
-    ``kept_mutations_before_end_time`` parameter.
+If adding mutations to a tree sequence which already contains them, the `keep` parameter
+controls whether existing mutations are kept or discarded (the default is `keep=True`).
+For instance, in final code block in {ref}`sec_mutations_time_span`, mutations were
+progressively added to a simulated tree sequence, beginninng with the oldest time period.
+
+While it is simple to add younger mutations to a tree sequence which already contains
+mutations, adding a mutation ancestral to an existing mutation will result in
+an error unless `add_ancestral=True` is specified, as in the following code block:
 
+```{code-cell}
+ts = msprime.sim_ancestry(5, random_seed=1)
+mts = msprime.sim_mutations(ts, rate=1, random_seed=1)
+mts = msprime.sim_mutations(mts, rate=0.1, random_seed=5, add_ancestral=True)
 ```
 
+Note that specifying `add_ancestral=True` does not guarantee that mutations
```

Comment:
I think we can add a ..note here mentioning that silent transitions *may* occur when you do this. We're making silent transitions legal in the upcoming tskit release. they are largely harmless, but something to be aware of.

(The code needs an update, but we might was well get the docs right now)